aten/src/ATen/cuda/CUDAGeneratorImpl.h

```diff
@@ -117,14 +156,13 @@ struct TORCH_CUDA_CPP_API CUDAGeneratorImpl : public c10::GeneratorImpl {
 
   static c10::DeviceType device_type();
 
-private:
+ private:
   CUDAGeneratorImpl* clone_impl() const override;
-  uint64_t seed_ = default_rng_seed_val;
-  uint64_t philox_offset_per_thread_ = 0;
-  int64_t* seed_extragraph_{};
-  int64_t* offset_extragraph_{};
-  uint32_t offset_intragraph_ = 0;
-  bool graph_expects_this_gen_ = false;
+  CUDAGeneratorState& state();
+  const CUDAGeneratorState& state() const;
+
+  size_t current_state_id_ = 0;
+  std::vector<CUDAGeneratorState> states_;
```

Comment:
In this segment of the code, we encapsulate the state of the generator within a structure named `CUDAGeneratorState`. The array `std::vector<CUDAGeneratorState>` consists of states for the CUDAGenerator.